src/app/AiPlayer.ts

```diff
@@ -0,0 +1,195 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { AnimalRoles } from '~src/Enums/AnimalRolesEnum';
+import { Player } from '../Player';
+import { GameController } from './GameController';
+import { Offer } from './Trade';
+import { ConvertAnimalName } from './utils/ConvertAnimalName';
+
+export class AiPlayer extends Player {
+  /**
+   * Makes a move for AI Player
+   * @param gameController for the game that the player is participating in
+   */
+  makeAMove(gameController: GameController): void {
+    this.trade(gameController);
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+    gameController.breed();
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
```

Comment:
Case like above. Please check

Reply:
After changes suggested above this if statements are very simple now (condition + return).